### DISCUSSION TITLE
Getting & observing properties like 'available_input_routing_channels'

### DISCUSSION THREAD

**Discussion by Gaasduif** [leolabs/ableton-js#83] [Q&A]
Track properties like 'available_input_routing_channels' return a string like '<Track.RoutingChannelVector object at 0x11891d2d0>'.
What is needed to get a js-object from this?

**Comment by leolabs** [comment]
ableton-js tries to automatically convert structures from Live to JSON objects, but more complex ones have to be implemented manually. I'll take a look at `available_input_routing_channels` soon.

**Reply by leolabs** [reply]
Just released a new version with support for:

- available_input_routing_channels
- available_input_routing_types
- available_output_routing_channels
- available_output_routing_types

Let me know if this works for you :)
